### PR TITLE
Adding netrc environment variable into the documentation

### PR DESCRIPTION
**Pull request by meist0731**
The starting point for contributions should usually be [a discussion](https://github.com/encode/httpx/discussions)

Simple documentation typos may be raised as stand-alone pull requests, but otherwise
please ensure you've discussed the your proposal prior to issuing a pull request.

This will help us direct work appropriately, and ensure that any suggested changes
have been okayed by the maintainers.

- [ ] Initially raised as discussion #...

### CODE REVIEW

File: docs/environment_variables.md [hunk 1]
```diff
@@ -175,3 +194,5 @@ python -c "import httpx; httpx.get('http://example.com')"
 python -c "import httpx; httpx.get('http://127.0.0.1:5000/my-api')"
 python -c "import httpx; httpx.get('https://www.python-httpx.org')"
 ```
+
+
```
Comment (tomchristie) [inline docs/environment_variables.md:198]: For tidiness we could do without the trailing newlines here.

File: docs/environment_variables.md [hunk 2]
```diff
@@ -132,6 +132,25 @@ Example:
 SSL_CERT_DIR=/path/to/ca-certs/ python -c "import httpx; httpx.get('https://example.com')"
 ```
 
+## `.NETRCFILE`
+
+Valid values: a filename
```
Comment (tomchristie) [inline docs/environment_variables.md:137]: "A filename"

File: docs/environment_variables.md [hunk 3]
```diff
@@ -132,6 +132,25 @@ Example:
...
+
+If this environment variable is set and auth parameter is not defined, HTTPX will add auth information stored in
+.netrc file into request's header. If you do not provide NETRC environment, HTTPX will use default files. (~/.netrc
+, ~/_netrc)
```
Comment (tomchristie) [inline docs/environment_variables.md:141]:
```suggestion
If this environment variable is set and auth parameter is not defined, HTTPX will add auth information stored in .netrc file into request's header. If you do not provide NETRC environment, HTTPX will use default files. (~/.netrc, ~/_netrc)
```

File: docs/environment_variables.md (hunk 1 above)
Comment (meist0731) [reply docs/environment_variables.md]: Get it. This makes more sense. Thank you for this good suggestion.

File: docs/environment_variables.md (hunk 3 above)
Comment (meist0731) [reply docs/environment_variables.md:141]: Yep. Couldn't agree more. This seems far better than mine. Thank you for this kind suggestion and thoughtful modification. 🙏🏻

**Comment by meist0731** [comment]
@tomchristie and @Vibhu-Agarwal, I've made changes to the description and the example part, but as for the 'Valid values: A filename', I changed it back to lowercase 'a', cuz other env variable descriptions all use the lowercase, if you need all be uppercase, do I need to create a new pull request to do so? Thanks in advance for checking them. :D

File: docs/environment_variables.md (hunk 3 above)
Comment (tomchristie) [reply docs/environment_variables.md]: Appreciating your style of interaction here, @meist0731. ✨

**Comment by tomchristie** [comment]
> as for the 'Valid values: A filename', I changed it back to lowercase 'a', cuz other env variable descriptions all use the lowercase

Ah, so they do, yup. Good call. We should prefer consistency in that case. If we do want to uppercase 'em then treat it as a separate PR yup. (Not *too* fussed about it, but yeah after a colon the following ought to be uppercased right?)

File: docs/environment_variables.md [hunk 4]
```diff
@@ -132,6 +132,18 @@ Example:
 SSL_CERT_DIR=/path/to/ca-certs/ python -c "import httpx; httpx.get('https://example.com')"
 ```
 
+## `.NETRCFILE`
```
Comment (j178) [inline docs/environment_variables.md:135]: I think this env name should be `NETRC`, right?

https://github.com/encode/httpx/blob/3231211aa364c0406737b78a45268e08c434f94c/httpx/_utils.py#L134-L135

and per the doc:
https://www.python-httpx.org/advanced/#netrc-support

File: docs/environment_variables.md [hunk 5]
```diff
@@ -132,6 +132,18 @@ Example:
...
+
+Valid values: a filename
+
+If this environment variable is set but auth parameter is not defined, HTTPX will add auth information stored in the .netrc file into the request's header. If you do not provide NETRC environment either, HTTPX will use default files. (~/.netrc, ~/_netrc)
+
+Example:
+
+```console
+NETRCFILE=/path/to/netrcfile/.my_netrc python -c "import httpx; httpx.get('https://example.com')"
```
Comment (j178) [inline docs/environment_variables.md:144]:
```suggestion
NETRC=/path/to/netrcfile/.my_netrc python -c "import httpx; httpx.get('https://example.com')"
```

**Review by j178 — CHANGES_REQUESTED** [review]

File: docs/environment_variables.md (hunk 4 above)
Comment (meist0731) [reply docs/environment_variables.md:135]: Yep! you're right, this should be 'NETRC', especially in the example part. I would like to confirm one thing that I noticed in the doc  (https://www.python-httpx.org/advanced/#netrc-support) both the right content table and central content title are written with a dot as the start, like ".netrc", so should I need to change this title to be ".NETRC" or simply "NETRC"?

Comment (meist0731) [reply docs/environment_variables.md:135]: > I think this env name should be `NETRC`, right?
> 
> https://github.com/encode/httpx/blob/3231211aa364c0406737b78a45268e08c434f94c/httpx/_utils.py#L134-L135
> 
> and per the doc:
> https://www.python-httpx.org/advanced/#netrc-support

For consistency, '.NETRC' would be better👀may be.

Comment (j178) [reply docs/environment_variables.md]: In this `Environment Variables` section, I think we should just use `NETRC`. 👍

Comment (meist0731) [reply docs/environment_variables.md]: Gotcha! Thanks for this suggestion.

**Review by j178 — APPROVED** [review]
LGTM 💯

**Review by florimondmanca — APPROVED** [review]
Thank you :-)

**Pull request merged**